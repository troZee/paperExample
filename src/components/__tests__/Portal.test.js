import * as React from 'react';
import { render, waitForElement } from 'react-native-testing-library';
import { Text } from 'react-native';
import Portal from '../Portal/Portal.tsx';

it('renders portal with siblings', async () => {
  const { toJSON, getByTestId } = render(
    <Portal.Host>
      <Text>Outside content</Text>
      <Portal>
        <Text testID="content">Portal content</Text>
      </Portal>
    </Portal.Host>
  );

  await waitForElement(() => getByTestId('content'));

  expect(toJSON()).toMatchSnapshot();
});
