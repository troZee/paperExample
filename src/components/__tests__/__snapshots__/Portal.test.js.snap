// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders portal with siblings 1`] = `
Array [
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text>
      Outside content
    </Text>
  </View>,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      testID="content"
    >
      Portal content
    </Text>
  </View>,
]
`;
